flappingangles.m

```diff
@@ -0,0 +1,110 @@
+% but WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+% General Public License for more details.
+% <http://www.gnu.org/licenses/>.
+%
+% ==============================================================================================
+% |Name        : flappingangles.m                                                        |
+% |Authors     : Matteo Cassetta                                                         |
+%                Pasquale Antonio Federico                                               |
+% |              University of Naples Federico II                                        |
+% |Version     : 1.0                                                                     |
+% |Date        : 08/01/2021                                                              |
+% |Modified    : 13/01/2021                                                              |
+% |Description : Rotor blade flapping angles                                             |
+% |Reference   : Prouty, R.W. - Helicopter Performance, Stability and Control            |
+% |Input       : Airspeed, angle of attack (deg), rotor/helicopter gross weight,         |
+% |              rotor Lock number, rotor solidity,rotor blade twist (deg,assumed linear)|
+% |              blade airfoil lift curve slope (1/rad) , rotor blade tip speed,         |
+% |              flapping hinge eccentricity/blade radius, blade radius;                 |
+% |              optional inputs allow choosing of the output mode                       |
+% |              and azimuth angle sample in "angles" mode                               |
+% |Output      : Azimuth angles psi, flapping angles beta and dbeta/dpsi                 |
+% |Note        :                                                                         |
+% ==============================================================================================
+
+function [varargout] = flappingangles(V,alpha,W,gamma,...
+    sigma,theta_tw,cla,omegaR,eR,R,options)
+
+% Argument validation
+arguments
+    V        (1,1) {mustBeNumeric,mustBeReal}
+    alpha    (1,1) {mustBeNumeric,mustBeReal}
+    W        (1,1) {mustBeNumeric,mustBeNonnegative}
+    gamma    (1,1) {mustBeNumeric,mustBeNonnegative}
+    sigma    (1,1) {mustBeNumeric,mustBeNonnegative}
+    theta_tw (1,1) {mustBeNumeric,mustBeReal}
+    cla      (1,1) {mustBeNumeric,mustBeReal}
+    omegaR   (1,1) {mustBeNumeric,mustBeNonnegative}
+    eR       (1,1) {mustBeNumeric,mustBeNonnegative}
+    R        (1,1) {mustBeNumeric,mustBeNonnegative}
+    options.output  (1,1) string {mustBeMember(options.output,{'angles','coefficients'})} = 'angles'
+    options.sample  (:,1) {mustBeNumeric,mustBeReal} = linspace(0,360,200)'
+    
+end
+
+%% Preliminary calculations
+
+% Converting to radians
+theta_tw = theta_tw*pi/180;
+
+% Thrust coefficient
+Tc = W/(1.225*omegaR^2*pi*R^2);

```

Comment:
Instead of `1.225` use a variable `rho`.

Reply:
Should I add the altitude to the inputs and use it to obtain the density or just put rho = 1.225?

Reply:
It is up to you, because the altitude will be one of the variables in the input file. So that this information will be taken using the IO functions.

Reply:
I added the altitude as a variable and put a value of 0 for now, to be updated when the IO functions are ready.